src/domain/Customer.ts

```diff
@@ -0,0 +1,52 @@
+import { NotEligibleForMortgageException } from '../exceptions/NotEligibleForMortgageException';
+
+export interface ICustomer {
+    id: number;
+    firstName: string;
+    lastName: string;
+    balance: number;
+    badCreditHistoryCount: number;
+    updateBalance?: (amount: number) => void;
+
+}
+
+type CustomerParams = Omit<ICustomer, 'id'> & { id: number };
+
+class Customer implements ICustomer {
+    id: number;
+    firstName: string;
+    lastName: string;
+    balance: number;
+    badCreditHistoryCount: number;
+
+    constructor({ id, firstName, lastName, balance, badCreditHistoryCount }: CustomerParams) {
+        this.id = id;
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.balance = balance;
+        this.badCreditHistoryCount = badCreditHistoryCount;
+    }
```

Comment:
We can simplify this part
```suggestion
    constructor(
        public id: number,
        public firstName: string,
        public lastName: string,
        public balance: number,
        public badCreditHistoryCount: number
    ) {}
```